### DISCUSSION TITLE
Handle AutoDispose Notifier dispose process manually problem

### DISCUSSION THREAD

**Discussion by thisisyusub** [rrousselGit/riverpod#1734] [Q&A] [answered]
Hi. in Riverpod 1.x.x I have tried to create notifier, which is auto disposed, but I am keeping state until user logOut from account.

<img width="455" alt="Screen Shot 2022-10-05 at 20 01 08" src="https://user-images.githubusercontent.com/44983544/194107420-b34251ed-3fb3-4526-afc3-a47493e42f23.png">
<img width="683" alt="Screen Shot 2022-10-05 at 20 00 49" src="https://user-images.githubusercontent.com/44983544/194107500-856cead5-eead-4170-ac80-5190529985da.png">

It was worked nicely what I want.  But after Riverpod 2.x.x I have changed it to the following.  But  in this case, first it dispose provider and then call method of notifier immediately what I have provided. That is why, it gets error because already user unauthenticated and give 401. Then I am login again and it will never call the same method again.

<img width="485" alt="Screen Shot 2022-10-05 at 20 04 24" src="https://user-images.githubusercontent.com/44983544/194108073-29601f75-818f-4bc5-b934-199431a1be43.png">

I don't understand "What's the problem here?", How can I do the same `maintainState = false` in Riverpod 2.0?
Thanks in advance.

**Comment by rrousselGit** [comment, marked as answer]
The equivalent of:

```dart
ref.maintainState = true;

ref.listen(..., (prev, next) {
  ref.maintainState = false;
});
```

is:


```dart
final link = ref.keepAlive();

ref.listen(..., (prev, next) {
  link.cancel();
});
```

**Reply by thisisyusub** [reply]
Thanks a lot! It is working.
